integration-tests/src/test/scala/tech/beshu/ror/integration/suites/RorStartingSuite.scala

```diff
@@ -0,0 +1,189 @@
+            .runSyncUnsafe(5 minutes)
+        }
+        "failed to load ROR ACL" in {
+          val esContainer = createEsContainer(
+            rorConfigFile = validRorConfigFile,
+            additionalEsYamlEntries = Map(notStartedResponseCodeKey -> "200") // unsupported response code
+          )
+
+          notStartedYetTestScenario(esContainer = esContainer, expectedResponseCode = 403)
+            .runSyncUnsafe(5 minutes)
+        }
+      }
+      "return not started response with http code 503" when {
+        "503 configured" in {
+          val esContainer = createEsContainer(
+            rorConfigFile = validRorConfigFile,
+            additionalEsYamlEntries = Map(notStartedResponseCodeKey -> "503")
+          )
+
+          notStartedYetTestScenario(esContainer = esContainer, expectedResponseCode = 503)
+            .runSyncUnsafe(5 minutes)
+        }
+      }
+    }
+  }
+
+  private def notStartedYetTestScenario(esContainer: EsContainer, expectedResponseCode: Int) = {
+    Task
+      .gatherUnordered(
+        List(
+          startContainer(esContainer),
+          testTrafficAndStopContainer(esContainer, expectedResponseCode)
+        )
+      )
+  }
+
+  private def testTrafficAndStopContainer(esContainer: EsContainer, expectedResponseCode: Int): Task[Unit] = {
+    for {
+      restClient <- createRestClient(esContainer)
+      searchTestResults <- searchTest(client = restClient, searchAttemptsCount = 100000)
+      _ <- stopContainer(esContainer)
+      result <- handleResults(searchTestResults, expectedResponseCode)
+    } yield result
+  }
+
+  private def searchTest(client: RestClient, searchAttemptsCount: Int): Task[Seq[TestResponse]] = Task.delay {
+    val sm = new SearchManager(client)
+    Range.inclusive(1, searchAttemptsCount).flatMap { _ =>
+      Try(sm.searchAll("*")) match {
+        case Failure(_) =>
+          None
+        case Success(response) =>
+          Some(TestResponse(response.responseCode, response.responseJson))
+      }
+    }
+      .distinct
+  }
+
+  private def handleResults(results: Seq[TestResponse], expectedResponseCode: Int): Task[Unit] = {
+    val hasEsRespondedWithNotStartedResponse = results
+      .filter { response =>
+        response.responseCode == expectedResponseCode
+      }
+      .exists { response =>
+        response
+          .responseJson("error")
+          .obj("root_cause")
+          .arr.exists { json =>
+          json("reason").str == "forbidden" &&
+            json("due_to").str == "READONLYREST_NOT_READY_YET"
+        }
+      }
+
+    val hasEsRespondedWithSuccess = results.exists(_.responseCode == 200)
+
+    if (hasEsRespondedWithNotStartedResponse && hasEsRespondedWithSuccess) {
+      Task.unit
+    } else {
+      Task.raiseError(new IllegalStateException(s"Test failed. Expected success response and ROR failed to start response but was: [$results]"))
+    }
+  }
+
+  private def createEsContainer(rorConfigFile: String,
```

Comment:
can we encapsulate all the container-related logic into a well-named private class?